### DISCUSSION TITLE
Merge two or more .speclib files

### DISCUSSION THREAD

**Discussion by brvpuyve** [vdemichev/DiaNN#74] [Q&A]
Hi Vadim,

Is there an option to merge two or more .speclib files within DIA-NN?

Best,
Bart

**Comment by vdemichev** [comment]
Hi Bart,

There will be in the next version. If you'd like to get it now, please email me, I can send you a development version which can do that. 

Best wishes,

Vadim

**Reply by haianhle** [reply]
Hi Vladim,

Is the current version 1.8 able to do this yet? If so how do I concatenate spectral libraries? Does DIANN print out the concatenated library? Thank you so much.

Anh

**Comment by hguturu** [comment]
Follow on to this question - Is there a way to export speclib to .tsv or another readable format? We are interested in making some modifications (such as remove some peptides or eventually replace peptide spectra).

**Reply by vdemichev** [reply]
Yes, just specify .speclib as input and .tsv as output library.

**Reply by hguturu** [reply]
I am trying `C:\DIA-NN\1.8\DiaNN.exe --verbose --lib .\1.8\ref\report-lib.predicted.speclib --out-lib .\1.8\ref\report-lib.predicted.tsv`, but it doesn't seem to be doing anything (i.e. no output). Could you clarify how to define the input/output or if there are other flags needed to do the conversion?

**Reply by hguturu** [reply]
Looks like it works with a dummy `--out dummy.tsv` file defined. `C:\DIA-NN\1.8\DiaNN.exe --verbose 1 --lib  .\1.8\ref\report-lib.predicted.speclib --out-lib .\1.8\ref\report-lib.predicted.tsv --gen-spec-lib --out dummy.tsv`

**Reply by vdemichev** [reply]
Yes, --gen-spec-lib was missing. But you can always use the GUI for this. When using a command line, please just copy the command which is beingcomposed by the GUI
